L1Trigger/GlobalMuonTrigger/interface/L1MuGlobalMuonTrigger.h

```diff
@@ -116,9 +116,9 @@ class L1MuGlobalMuonTrigger : public edm::one::EDProducer<edm::one::SharedResour
   bool m_writeLUTsAndRegs;
   bool m_sendMipIso;
 
-  static std::shared_ptr<L1MuGMTConfig> m_config;
+  static std::atomic<std::shared_ptr<L1MuGMTConfig>> m_config;
 
-  static std::shared_ptr<L1MuGMTDebugBlock> m_db;
+  static std::atomic<std::shared_ptr<L1MuGMTDebugBlock>> m_db;
```

Comment:
Why do these `static`s need to be treated atomically in the first place? The module is of type `edm::one::EDProducer` with the `SharedResources`, and the constructor has
https://github.com/cms-sw/cmssw/blob/66f3d2f3d52e8f8d8c79a9ebe2cd0542c33e1380/L1Trigger/GlobalMuonTrigger/src/L1MuGlobalMuonTrigger.cc#L131

so framework should serialize the calls to all instances of the module.


(although it would be much better if these member data would not be `static`s so module type could be more thread-efficient than `edm::one`)

Reply:
Well ok, the atomics were added in https://github.com/cms-sw/cmssw/pull/36965 (the actual reasons being somewhat obscure), so better to keep them.

Reply:
Looking at the usage of `DebugBlockForFill` even though the creating of the value is safe, lots of code sets the values which is NOT done in a thread safe manner.

I think both of these statics should just be removed.